crates/ruff_linter/src/message/mod.rs

```diff
@@ -83,6 +77,13 @@ impl MessageKind {
 }
 
 impl Message {
+    pub fn syntax_error(message: String, range: TextRange, file: SourceFile) -> Message {
```

Comment:
It might be a little nicer to accept a `impl std::fmt::Display` here (which is what most of the `Diagnostic` APIs accept, at least, in spirit). Then you could avoid the various `err.to_string()` calls and just pass `err`.